.github/workflows/ci-cd.yml

```diff
@@ -0,0 +1,91 @@
+name: CI/CD
+
+on:
+  push:
+    branches: [ main ]
+    tags: [ v* ]
+  pull_request:
+    branches: [ main ]
+  release:
+    types: [published]
+
+jobs:
+  unit-testing:
+    name: Unit Testing
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    - name: Use Node.js 12.x
+      uses: actions/setup-node@v1
+      with:
+        node-version: 12.x
+
+    - name: Install dependencies
+      run: npm ci
+
+    - name: Run linting
+      run: npm run lint
+
+    - name: Run tests
+      run: npm run test-ci
+
+    - name: Upload Test Coverage
+      uses: actions/upload-artifact@v2
+      with:
+        name: coverage
+        path: coverage/
+
+    # GitHub actions only provides a read-only token for PRs from forks which prevents comments
+    #- name: Comment Test Coverage
+    #  if: github.event_name == 'pull_request'
+    #  uses: romeovs/lcov-reporter-action@v0.2.11
+    #  with:
+    #    github-token: ${{ secrets.GITHUB_TOKEN }}
+
+  code-analysis:
+    name: Code Analysis
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
+      with:
+        languages: javascript
+
+    - name: Perform CodeQL Analysis
+      uses: github/codeql-action/analyze@v1
+
+  deployment:
+    if: startsWith(github.ref, 'refs/tags/v') || github.event_name == 'release'
```

Comment:
does this deploy on tags and releases? so there are two release when a tagged release is created?

Reply:
The idea I currently had in mind was to have a "stage" version that is deployed when a tag is set and the "production" one that is only deployed on an explicit release.
So far not really sure, to be honest, if this makes sense or works at all.